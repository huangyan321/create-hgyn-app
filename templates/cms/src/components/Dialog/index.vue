<template>
    <el-dialog
        v-model="show"
        class="form-dialog"
        :title="title"
        :width="width"
        :append-to-body="appendToBody"
        draggable
        @close="close"
    >
        <slot name="body"></slot>
        <slot name="footer"></slot>
    </el-dialog>
</template>

<script lang="ts" setup>
import { ref, watch } from 'vue';
const props = withDefaults(
    defineProps<{
        formShow?: boolean;
        title?: string;
        width?: string | number;
        appendToBody?: boolean;
    }>(),
    {
        formShow: false,
        title: '编辑信息',
        width: '65%',
        appendToBody: false,
    },
);
const show = ref(false);
watch(props, (n) => {
    show.value = n.formShow;
});
const emit = defineEmits(['close']);
const close = () => {
    emit('close');
};
</script>
