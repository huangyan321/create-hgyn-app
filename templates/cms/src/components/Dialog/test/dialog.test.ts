import { mount } from '@vue/test-utils';
import Dialog from '@/components/Dialog/index.vue';
import { expect, it, describe } from 'vitest';

describe('Dialog', () => {
    const wrapper = mount(Dialog, {
        props: {
            title: '编辑表单',
            width: '70%',
        },
    });
    it('snapeShot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('props', () => {
        expect(wrapper.props().title).toBe('编辑表单');
        expect(wrapper.props().width).toBe('70%');
    });
});
