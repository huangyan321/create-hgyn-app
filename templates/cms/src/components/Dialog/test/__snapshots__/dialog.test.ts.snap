// Vitest Snapshot v1

exports[`Dialog > snapeShot 1`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"el-overlay\\" style=\\"z-index: 2001; display: none;\\">
    <div role=\\"dialog\\" aria-modal=\\"true\\" aria-label=\\"编辑表单\\" aria-describedby=\\"el-id-1411-0\\" class=\\"el-overlay-dialog\\"></div>
  </div>
</transition-stub>
<!--teleport end-->"
`;
